controllers/noobaa-rules.yaml

```
@@ -1,104 +1,109 @@
          labels:
            origin: data-federation-service
-         record: job:noobaa_total_unhealthy_namespace_buckets
+         record: namespace:noobaa_unhealthy_namespace_buckets:max
        - expr: |
-           NooBaa_num_namespace_buckets{namespace="redhat-data-federation"}
+           max without(instance) (NooBaa_accounts_num{namespace="redhat-data-federation"})
          labels:
            origin: data-federation-service
-         record: job:noobaa_namespace_bucket_count
+         record: namespace:noobaa_accounts:max
        - expr: |
-           NooBaa_accounts_num{namespace="redhat-data-federation"}
+           max without(instance) (NooBaa_total_usage{namespace="redhat-data-federation"})
          labels:
            origin: data-federation-service
-         record: noobaa_accounts_num
+         record: namespace:noobaa_usage:max
        - expr: |
-           NooBaa_total_usage{namespace="redhat-data-federation"}
+           max without(instance) (NooBaa_num_unhealthy_namespace_resources{namespace="redhat-data-federation"})
          labels:
            origin: data-federation-service
-         record: noobaa_total_usage
+         record: namespace:noobaa_unhealthy_namespace_resources:max
        - expr: |
-            NooBaa_num_unhealthy_namespace_resources{namespace="redhat-data-federation"}
+           max without(instance) (NooBaa_num_namespace_resources{namespace="redhat-data-federation"})
          labels:
            origin: data-federation-service
-         record: noobaa_num_unhealthy_namespace_resources
+         record: namespace:noobaa_namespace_resources:max
        - expr: |
-           NooBaa_num_namespace_resources{namespace="redhat-data-federation"}
+           count without (instance) (NooBaa_namespace_resource_status{namespace="redhat-data-federation"} == 1)
          labels:
            origin: data-federation-service
-         record: noobaa_num_namespace_resources
+         record: namespace:healthy_noobaa_namespace_resources:count
        - expr: |
-           NooBaa_namespace_resource_status{namespace="redhat-data-federation"}
+           count without (instance) (NooBaa_namespace_resource_status{namespace="redhat-data-federation"} == 0)
          labels:
            origin: data-federation-service
-         record: noobaa_namespace_resource_status
+         record: namespace:unhealthy_noobaa_namespace_resources:count
        - expr: |
-           NooBaa_namespace_bucket_status{namespace="redhat-data-federation"}
+           max without(instance) (NooBaa_num_namespace_buckets{namespace="redhat-data-federation"})
          labels:
            origin: data-federation-service
-         record: noobaa_namespace_bucket_status
+         record: namespace:noobaa_namespace_buckets:max
+       - expr: |
+           count without (instance) (NooBaa_namespace_bucket_status{namespace="redhat-data-federation"} == 1)
```

IIUC there's a `bucket_name` label which can have multiple values. To have a cardinality of 1 it needs to be added to the without() clause:

```suggestion
           count without(instance, bucket_name) (NooBaa_namespace_bucket_status{namespace="redhat-data-federation"} == 1)
```

or to avoid any risk of cardinality

```suggestion
           count(NooBaa_namespace_bucket_status{namespace="redhat-data-federation"} == 1)
```

---

In general it is more convenient to count "all items" (a) and "items in error" (e) rather "good items" (g) and "items in error" (e) because you can compute the ratio of failed items more easily (e.g. e/a rather then e/(e+g)).

---

Made changes based on the second approach above/